packages/bundler/src/Bundler.ts

```diff
@@ -105,27 +133,83 @@ export class Bundler implements IBundler {
       userOpHash: sendUserOperationResponse.result,
       wait: (confirmations?: number): Promise<UserOpReceipt> => {
         const provider = new JsonRpcProvider(RPC_PROVIDER_URLS[chainId]);
+        // Note: maxDuration can be defined per chainId
+        const maxDuration = this.UserOpReceiptMaxDurationIntervals[chainId] || 30000; // default 30 seconds
+        let totalDuration = 0;
+
         return new Promise<UserOpReceipt>((resolve, reject) => {
-          const intervalId = setInterval(async () => {
-            try {
-              const userOpResponse = await this.getUserOpReceipt(sendUserOperationResponse.result);
-              if (userOpResponse && userOpResponse.receipt && userOpResponse.receipt.blockNumber) {
-                if (confirmations) {
-                  const latestBlock = await provider.getBlockNumber();
-                  const confirmedBlocks = latestBlock - userOpResponse.receipt.blockNumber;
-                  if (confirmations >= confirmedBlocks) {
+          if (this.UserOpReceiptIntervals && chainId in this.UserOpReceiptIntervals) {
+            const intervalValue = this.UserOpReceiptIntervals[chainId];
+            if (intervalValue !== undefined) {
+              const intervalId = setInterval(async () => {
+                try {
+                  const userOpResponse = await this.getUserOpReceipt(sendUserOperationResponse.result);
+                  if (userOpResponse && userOpResponse.receipt && userOpResponse.receipt.blockNumber) {
+                    if (confirmations) {
+                      const latestBlock = await provider.getBlockNumber();
+                      const confirmedBlocks = latestBlock - userOpResponse.receipt.blockNumber;
+                      if (confirmations >= confirmedBlocks) {
+                        clearInterval(intervalId);
+                        resolve(userOpResponse);
+                      }
+                    }
                     clearInterval(intervalId);
                     resolve(userOpResponse);
                   }
+                } catch (error) {
+                  clearInterval(intervalId);
+                  reject(error);
+                }
+
+                totalDuration += intervalValue;
+                if (totalDuration >= maxDuration) {
+                  clearInterval(intervalId);
+                  reject(new Error("Exceeded maximum duration"));
```

Comment:
Change the message to 
`Exceeded maximum duration (${maxDuration/1000} sec) waiting to get receipt for userOpHash ${sendUserOperationResponse.result}. Try getting the receipt manually using eth_getUserOperationReceipt rpc method on bundler`